### ISSUE TITLE
Intersection of eigenvalues

### ISSUE THREAD

**Issue by galvanetto** [sbyrnes321/cf#1]
Dear SB,
in your wonderful https://coldfusionblog.net/2017/07/09/numerical-spin-boson-model-part-1/ I have a concern with your plot number 1: the colored lines are supposed to intersects, but in your case they simply touch each other maintaining their position.

I think the problem is due to the incorrect ordering of the eigenvalues, i.e. what was the i-th eigenvalue at g=0 may became the j-th eigenvalue at coupling g!=0.

Can you comment on this
Best

**Comment by sbyrnes321** [comment]
Thanks for your comment.

I believe that the curves [don't actually cross](https://en.wikipedia.org/wiki/Avoided_crossing), although they get awfully close to crossing. So my color scheme follows an eigenvalue whose eigenvector is continuously changing along the curve. For your proposed color scheme, we would swap the colors (discontinuously) somewhere around each anticrossing. Sure, that also works. Or we could color all the curves black. It's just a color scheme—aesthetics and pedagogy, there isn't a right or wrong answer. I don't plan to change it, sorry. :-P

**Comment by galvanetto** [comment]
Dear SB,

sorry, you are right and I made a mistake [reproducing your code](https://github.com/project-ida/two-state-quantum-systems/blob/nicola-sandbox/sjb-qutip.ipynb) with the QuTip python library, which did not separate between odd and even states. It resulted in some true crossing and some true anticrossing. I solved the problem [here](https://github.com/project-ida/two-state-quantum-systems/blob/nicola-sandbox/sjb-qutip.ipynb) in the last plots, looking for continuity in the eigenstates.

Thank you for your work!
Nicola

![Cattura](https://user-images.githubusercontent.com/28784964/74380779-86f34f80-4dea-11ea-82f3-d2572fe2f2a0.JPG)
